### DISCUSSION TITLE
LSP rename not responding until cursor move due to `expr = true`?

### DISCUSSION THREAD

**Discussion by briandipalma** [LazyVim/LazyVim#220] [General]
The LSP rename is acting odd compared to what I'm used to. I can't tell if it's just me or it's meant to be like that.
The input for the new name won't always show up immediately. The first rename it usually does but the second onwards seems to require a cursor move to show. The issue is fixed when I remove

`expr = true`

from

`{ "<leader>cr", M.rename, expr = true, desc = "Rename", has = "rename" },`

Is this a bug, am I missing something, am I the only person seeing this?

**Comment by dpetka2001** [comment]
It happens to me occasionally as well. It seems to me as when I use `h,j,k,l` to navigate, some times they must be buffered and considered as keycodes when I invoke the `lsp-rename`. If i wait a bit on the word I want to rename (say 1-2 secs) it doesn't seem to happen. It only happens after I move to a word and immediately invoke `lsp-rename`. That's only my assumption, let's wait for the maintainer's thoughts as well. It doesn't bother me that much to be honest.

**Comment by folke** [comment]
I just pushed a fix for this. Will make a new release soon. Thank you for letting me know!
